components/base/wheeled/wheeled_base.go

```diff
@@ -115,6 +133,97 @@ func (base *wheeledBase) Spin(ctx context.Context, angleDeg, degsPerSec float64,
 	rpm, revolutions := base.spinMath(angleDeg, degsPerSec)
 
 	return base.runAll(ctx, -rpm, revolutions, rpm, revolutions)
+
+}
+
+func (base *wheeledBase) spinWithMovementSensor(ctx context.Context, angleDeg, degsPerSec float64, extra map[string]interface{}) error {
+	wheelrpm, revs := base.spinMath(angleDeg, degsPerSec)
+
+	currYaw, err := getCurrentYaw(ctx, base.orientationSensor, extra) // from -179 to 179
+	if err != nil {
+		return err
+	}
+
+	errCounter := 0
+	targetYaw := addAnglesInDomain(angleDeg, currYaw)
+	timer := time.NewTicker(time.Duration(yawPollTimeMs * float64(time.Millisecond))) //~rename
+
+	if err := base.runAll(ctx, -wheelrpm, revs, wheelrpm, revs); err != nil {
```

Comment:
If I understand the flow correctly runAll activates GoFor for each motor[s]. And each GoFor will run for n seconds or  until a target is reached. 
So when runAll returns supposedly all GoFor would have completed (except if context is cancelled in this case it seems somewhat UB) and the base shouldn't be moving anymore.  If this is right does the for loop helps?

Reply:
The other design of this is to run `runAll` with increments calculated on each loop, the problem with that is that the underlying `RPMMonitor` in `GoFor` can't give you smooth motion because of the start/stop behavior of that driver. I'm going to try out both today, and decide what improvements to make to the motor performance.

Reply:
they are now both in ogroutines that function!